app/models/concerns/profile_rules.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+# Methods that are related to a profile's rules
+module ProfileRules
```

Comment:
This and ProfileHosts modules were separated out of the Profile model due to class length restrictions. Also, there's quite a bit of logic here for ProfileRules; it's nice having it in one place.